docs/developer/payments/transactions.mdx

```diff
@@ -450,6 +450,80 @@ Once `Checkout` is converted to into `Order`, the `authorizeStatus` might change
 - **For orders**: pending Transactions **are not** considered "paid" and will not be included in the [`Order.authorizeStatus`](/api-reference/orders/enums/order-authorize-status-enum.mdx#values) calculation.
 :::
 
+### TransactionItem and  TransactionEvents pspReference Matching
+
+Both [TransactionItem](/api-reference/payments/objects/transaction-item)
+and [TransactionEvent](/api-reference/payments/objects/transaction-event)
+contain a `pspReference` field, which refers to the payment service provider reference.
+	-	The `pspReference` on `TransactionItem` identifies the overall payment.
+	-	The `pspReference` on `TransactionEvent` identifies specific actions that occurred within the `Transaction`.
+
+**Key Assumptions:**
+- Only one `TransactionItem` instance can exist with a given `pspReference`.
+- For a specific `TransactionItem`, only one `TransactionEvent` of a given `type` and `pspReference` may exist, except for the following types:
+  -	`AUTHORIZATION_ACTION_REQUIRED`
+	-	`CHARGE_ACTION_REQUIRED`
+	-	`INFO`
+- Only one `TransactionEvent` of type `AUTHORIZATION_SUCCESS` is allowed to exist.
+
+#### Setting the `pspReference` on the `TransactionItem`
+
+The pspReference value in a TransactionItem may initially be empty if it is not provided
+during the TransactionCreate mutation or as a result of the TransactionInitialize mutation.
+When the first `TransactionEvent` containing a `pspReference` is reported,
+this value is automatically set on the associated `TransactionItem`.
+
+The `pspReference` can be updated through the `transactionUpdate` mutation if necessary.
+
+#### Events with optional `pspReference` 
```

Comment:
We have a section: https://docs.saleor.io/developer/payments/transactions#amount-calculations-for-reporting-an-action-with-missing-amount-value were we describe when amount can be optional. Isn't like this should also be there? This section described `Reporting actions for transactions`. Right now some info about optional fields is there, and some info is here 🤔 

Also, what about the "webhook-part": https://docs.saleor.io/developer/extending/webhooks/synchronous-events/transaction#response-4
Don't we need to also say there that the reference can be optional for specific events?

Reply:
> Also, what about the "webhook-part": https://docs.saleor.io/developer/extending/webhooks/synchronous-events/transaction#response-4
> Don't we need to also say there that the reference can be optional for specific events?

From what I see it's already mentioned. 
![image](https://github.com/user-attachments/assets/772ed087-921b-4922-a05c-18f79b883407)
But looks like the list is not full, so I will extend it.